### DISCUSSION TITLE
How to use coc.nivm to replace ale's grammar check and vim-autoformat's automatic formatting.

### DISCUSSION THREAD

**Discussion by sumengmsm** [neoclide/coc.nvim#3064] [Q&A] [answered]
I use the plug, coc.pyright. There seems to be this function in the document, but there is no detailed description. I am a new vimer, so please bear with me.

**Comment by chemzqm** [comment, marked as answer]
Don't know, ale has many built-in formatters, but you need language server or extension when using coc.nvim.
